official/transformer/transformer_main.py

```diff
@@ -69,37 +73,71 @@ def model_fn(features, labels, mode, params):
     # Create model and get output logits.
     model = transformer.Transformer(params, mode == tf.estimator.ModeKeys.TRAIN)
 
-    output = model(inputs, targets)
+    logits = model(inputs, targets)
 
     # When in prediction mode, the labels/targets is None. The model output
     # is the prediction
     if mode == tf.estimator.ModeKeys.PREDICT:
+      if params["use_tpu"]:
+        raise NotImplementedError("Prediction is not yet supported on TPUs.")
```

Comment:
Should this `NotImplementedError` be raised before the line `logits = model(inputs, targets)`?

Reply:
It should be fine here. Just defining the ops isn't an issue, and I like to keep it close to the non-TPU PREDICT case.